ding_spt_statistics.module

```diff
@@ -3,3 +3,153 @@
+    'page callback' => 'ding_spt_statistics_display',
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_permission().
+ */
+function ding_spt_statistics_permission() {
+  return array(
+    'view spt statistics' => array(
+      'title' => t('View SPT statistics'),
+      'description' => t('View statistics for clicked external SPT resources.'),
+    ),
+  );
+}
+
+/**
+ * Count click and redirect to the resource.
+ */
+function ding_spt_statistics_redirect() {
+  // Register statistical data.
+  $hostname = !empty($_GET['hostname']) ? $_GET['hostname'] : FALSE;
+  ding_spt_statistics_register($hostname);
+
+  // Redirect to the external resource.
+  $path = !empty($_GET['path']) ? $_GET['path'] : FALSE;
+  drupal_goto($path);
+}
+
+/**
+ * Register hostname into the statistics table.
+ *
+ * @param $hostname string
+ *   Name of the external host.
+ */
+function ding_spt_statistics_register($hostname) {
+  if (!$hostname) {
+    return;
+  }
+  $result = db_select('ding_spt_statistics', 'dss')
+    ->fields('dss', array('hid', 'count'))
+    ->condition('hostname', $hostname)
+    ->execute()
+    ->fetchAssoc()
+  ;
+
+  // Update table statistics with new values.
+  if (!$result) {
+    $result['hid'] = db_insert('ding_spt_statistics')
+      ->fields(array(
+        'hostname' => $hostname,
+        'count' => 1,
+      ))
+      ->execute();
+  }
+  else {
+    db_update('ding_spt_statistics')
+      ->expression('count', 'count + 1')
+      ->condition('hostname', $hostname)
+      ->execute();
+  }
+
+  // Insert new record to the data table.
+  db_insert('ding_spt_hits')
+    ->fields(array(
+      'hid' => $result['hid'],
+      'accessed' => REQUEST_TIME,
+    ))
+    ->execute();
+}
+
+/**
+ * Display statistics of the hits as a table.
+ *
+ * @return string
+ *   Themed table.
+ */
+function ding_spt_statistics_display() {
+  // Get data from db.
+  $query = db_select('ding_spt_statistics', 's');
+  $query->leftJoin('ding_spt_hits', 'h', 's.hid = h.hid');
+  $query->fields('s', array('hostname', 'count'));
+
+  $query->addExpression('SUM(CASE WHEN unix_timestamp(now()) - h.accessed < 86400 THEN 1 ELSE 0 END)', 'day');
+  $query->addExpression('SUM(CASE WHEN unix_timestamp(now()) - h.accessed < 7*86400 THEN 1 ELSE 0 END)', 'week');
+  $query->addExpression('SUM(CASE WHEN unix_timestamp(now()) - h.accessed < 30*86400 THEN 1 ELSE 0 END)', 'month');
+  $query->addExpression('SUM(CASE WHEN unix_timestamp(now()) - h.accessed < 90*86400 THEN 1 ELSE 0 END)', '3months');
+
+  $query->groupBy('s.hid');
+  $result = $query->execute()->fetchAllAssoc('hostname');
+
+  // Build markup for the table.
+  $rows = array();
+  foreach ($result as &$item) {
+    $rows[] = array(
+      l($item->hostname, 'http://' . $item->hostname, array('attributes' => array('target'=>'_blank'))),
+      $item->day,
+      $item->week,
+      $item->month,
+      $item->{'3months'},
+      $item->count,
+    );
+  }
```

Comment:
Will be nice to see here 
1) default order, as example by hostname
2) sortable tableheader
3) pagination